boa/src/exec/switch/mod.rs

```diff
@@ -0,0 +1,34 @@
+use super::{Executable, Interpreter};
+use crate::{
+    builtins::value::{ResultValue, Value},
+    syntax::ast::node::Switch,
+};
+
+impl Executable for Switch {
+    fn run(&self, interpreter: &mut Interpreter) -> ResultValue {
+        let default = self.default();
+        let val = self.val().run(interpreter)?;
+        let mut result = Value::null();
+        let mut matched = false;
+        for tup in self.cases().iter() {
+            let cond = &tup.0;
+            let block = &tup.1;
+            if val.strict_equals(&cond.run(interpreter)?) {
+                matched = true;
+                let last_expr = block.last().expect("Block has no expressions");
+                for expr in block.iter() {
+                    let e_result = expr.run(interpreter)?;
+                    if expr == last_expr {
+                        result = e_result;
+                    }
+                }
```

Comment:
The block itself should be `Executable`, right? so we could just run it.

Reply:
Won't this have problems when it comes to implementing break because if we execute block then how do we know if it ended with break and therefore should immediately return from switch.run()?

Reply:
Breaks should be implemented the same way to a return, so we should be able to do that here.

Reply:
Ah ok so with a flag?

Reply:
> Ah ok so with a flag? 

Yep, in the interpreter. At some point there will be flags for awaiting, yielding and more.